### ISSUE TITLE
FHIR Search using Reference Behavior working in unexpected ways

### ISSUE THREAD

**Issue by prb112** [IBM/FHIR#1570]
**Describe the bug**
Reference Search Behavior working in unexpected ways

**To Reproduce**
1. Create a Patient
```
curl --location --request POST 'https://localhost:9443/fhir-server/api/v4/Patient' \
--header 'Authorization: Basic CHANGEME' \
--header 'Content-Type: application/json' \
--data-raw '{
  "resourceType": "Patient",
  "managingOrganization": {
       "reference": "Organization/1",
        "display": "Demo Org"
  },
  "contact": [
    {
      "relationship": [
        {
          "coding": [
            {
              "system": "http://terminology.hl7.org/CodeSystem/v2-0131",
              "code": "E"
            }
          ]
        }
      ],
      "organization": {
        "reference": "Organization/1",
        "display": "Demo Org"
      }
    }
  ],
  "text": {
    "status": "generated",
    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">\n      \n      <table>\n        \n        <tbody>\n          \n          <tr>\n            \n            <td>Name</td>\n            \n            <td>Peter James \n              <b>Chalmers</b> (&quot;Jim&quot;)\n            </td>\n          \n          </tr>\n          \n          <tr>\n            \n            <td>Address</td>\n            \n            <td>534 Erewhon, Pleasantville, Vic, 3999</td>\n          \n          </tr>\n          \n          <tr>\n            \n            <td>Contacts</td>\n            \n            <td>Home: unknown. Work: (03) 5555 6473</td>\n          \n          </tr>\n          \n          <tr>\n            \n            <td>Id</td>\n            \n            <td>MRN: 12345 (Acme Healthcare)</td>\n          \n          </tr>\n        \n        </tbody>\n      \n      </table>    \n    \n    </div>"
  },
  "identifier": [
    {
      "use": "usual",
      "type": {
        "coding": [
          {
            "system": "http://hl7.org/fhir/v2/0203",
            "code": "MR"
          }
        ]
      },
      "system": "urn:oid:1.2.36.146.595.217.0.1",
      "value": "12345",
      "period": {
        "start": "2001-05-06"
      },
      "assigner": {
        "display": "Acme Healthcare"
      }
    }
  ],
  "active": true,
  "name": [
    {
      "use": "official",
      "family": "Chalmers",
      "given": [
        "Peter",
        "James"
      ]
    },
    {
      "use": "usual",
      "given": [
        "Jim"
      ]
    }
  ],
  "telecom": [
    {
      "use": "home"
    },
    {
      "system": "phone",
      "value": "(03) 5555 6473",
      "use": "work"
    }
  ],
  "gender": "male",
  "birthDate": "1974-12-25",
  "_birthDate": {
    "extension": [
      {
        "url": "http://hl7.org/fhir/StructureDefinition/patient-birthTime",
        "valueDateTime": "1974-12-25T14:35:45-05:00"
      }
    ]
  },
  "deceasedBoolean": false,
  "address": [
    {
      "use": "home",
      "type": "both",
      "line": [
        "534 Erewhon St"
      ],
      "city": "PleasantVille",
      "district": "Rainbow",
      "state": "Vic",
      "postalCode": "3999",
      "period": {
        "start": "1974-12-25"
      }
    }
  ],
  "extension": [ 
    {
        "url": "http://com.ibm.fhir.sample/favorite-mlb",
        "valueString": "RedSox"
    },
    {
        "url": "http://com.ibm.fhir.sample/favorite-nfl",
        "valueString": "bad-bad"
    }
  ]
}'
```
2. Call ```curl --location --request GET 'https://localhost:9443/fhir-server/api/v4/Patient?organization=https://localhost:9443/fhir-server/api/v4/Organization/1' \
--header 'Authorization: Basic CHANGEME' \
--header 'Content-Type: application/json'```

It should return the Patient, however due to a limited regular expression it returns nothing. 

**Expected behavior**
It should return the Patient.

**Additional context**
n/a

**Comment by lmsurpre** [comment]
was this bug introduced with the fix for #300?  or were there parts of this that were wrong prior to that fix as well?  just asking for clarity reasons in the 4.4.1 release notes....if it was introduced in the fix for #300 then I don't think we need to mention the bug in our release notes at all...will just mention the new support introduced.

**Comment by prb112** [comment]
I introduced this bug with #300
